generator/src/main/java/com/scottlogic/deg/generator/generation/field_value_sources/IntegerFieldValueSource.java

```diff
@@ -112,11 +112,12 @@ public long getValueCount() {
         return () -> new UpCastingIterator<>(
             Stream.of(
                 incrementWhile(inclusiveLower, blacklist::contains),
-                inclusiveLower <= 0 && 0 < exclusiveUpper && !blacklist.contains(0)
+                inclusiveLower < 0 && 0 < exclusiveUpper && !blacklist.contains(0)
```

Comment:
now we're using distinct. we can replace the whole

inclusiveLower <= 0 && 0 < exclusiveUpper && !blacklist.contains(0)
                    ? 0
                    : null,

with just

0,